Feature/mobile

gochicus opened the pull request:
#74

gochicus committed:
WIP(web-diary): webpack config change

affects: @practice/web-diary

ISSUES CLOSED: #74

gochicus committed:
WIP(web-diary): media querries added

affects: @fragments/diary, @practice/web-diary, @ui/background

ISSUES CLOSED: #74

gochicus committed:
refactor(web-diary): markup refactor

affects: @fragments/diary, @practice/web-diary, @ui/layout, @ui/list

ISSUES CLOSED: #74

TFK70 commented on practice/web-diary/fragments/diary/src/Diary.tsx:
```diff
@@ -9,18 +9,22 @@ import { Box, Column, Layout, Row } from '@ui/layout'
 export const Diary = () => {
   return (
-    <Background gradient='blueAndGreenGradient' height='100vh' width='100vw'>
+    <Background gradient='blueAndGreenGradient' height='100vh' width={[1, 1, 1]}>
```
Ты так и не рассказал что за [1,1,1]

gochicus replied:
@TFK70 ну здесь можно просто на 100% заменить. А вот насчет того, что едет, так это всё резиновая верстка. Я точно не могу использовать своё решение с медиазапросом в бэкграунде?

TFK70 replied:
@gochicus 
1. [1,1,1] точно меняй на более адекватное значение ширины
2. Нет, это проблема в верстке

TFK70 reviewed (changes requested):
Верстка полетела (в том числе и на десктопе, отступы другие)
Вероятно, после добавления height='100%'

![IMG_8F55E9158B0F-1](https://user-images.githubusercontent.com/62806448/122053418-28841a00-ce00-11eb-8675-a5dd6a16248a.jpeg)

gochicus committed:
refactor(web-diary): markup refactored

affects: @fragments/diary, @practice/web-diary, @ui/list

ISSUES CLOSED: #74

TFK70 commented on practice/web-diary/fragments/diary/src/Diary.tsx:
```diff
@@ -29,7 +29,12 @@ export const Diary = () => {
         </Box>
-        <Layout flexBasis='45px' />
+        <Box
+          backgroundColor='white'
```
Даже медиа-запросы кажутся не такой плохой идеей на фоне этой
Пересмотри свой основной лейаут, мне все-таки как-то удалось добиться того же результата без подобных костылей (буквально одной строкой)

gochicus committed:
refactor(web-diary): markup refactor

affects: @fragments/diary, @practice/web-diary

ISSUES CLOSED: #74

TFK70 commented on practice/web-diary/fragments/diary/src/Diary.tsx:
```diff
@@ -1,18 +1,18 @@
       <Column alignItems='center'>
         <Layout flexBasis={['0px', '50px', '300px']} />
-        <Box backgroundColor='white' width={['100%', '400px', '400px']}>
+        <Column backgroundColor='white' width={['100%', '400px', '400px']} overflowY='auto'>
```
1. Зачем overflowY?
2. Ты видел десктоп версию?

gochicus replied:
@TFK70 overflow нужен, потому как при большом количестве постов они начинают выезжать за пределы белого блока.

gochicus replied:
@TFK70 overflow нужен, потому как при большом количестве постов они начинают выезжать за пределы белого блока.

gochicus committed:
refactor(web-diary): markup refactored

affects: @fragments/diary, @practice/web-diary

ISSUES CLOSED: #74

TFK70 commented on practice/web-diary/fragments/diary/src/Diary.tsx:
```diff
@@ -1,18 +1,18 @@
 export const Diary = () => {
   return (
-    <Background gradient='blueAndGreenGradient' height='100vh' width='100%'>
+    <Background gradient='blueAndGreenGradient' height={['50vh', '100vh', '100vh']} width='100%'>
```
Точно нет
Background всегда занимает все пространство от родителя, манипулировать его размерами таким образом нельзя, иначе это уже не бекграунд

TFK70 replied:
И из этого следует дефект:

https://user-images.githubusercontent.com/62806448/122208088-4e6ef480-cebc-11eb-85aa-2398dfdd4932.mov

gochicus replied:
@TFK70 Ну я бы не сказл, что это деффект. Если скроллить в белом блоке или в `column`,  то возникают проблемы с контентом, по сравнению с которыми это ерунда. А вообще в каком ключе  это делать? Уже третье решение не то, что надо. Так что надо?

TFK70 replied:
@gochicus я тебя немного не понимаю

1. Ты предлагаешь закрыть глаза на полуразваленную мобильную версию потому что по сравнению с полуразваленной десктопной это ерунда? 

2. > Так что надо?

Задача не понятна или готовый код нужен?

Сформулируй проблему, скажи что делал, приложи код, будем вместе смотреть что не так
И если тебе для этого придется некоторые куски приложения заново переписать - это нормально, до этого у тебя не было представления как должна выглядеть верстка с которой не будет проблем на мобилке. Сейчас есть

gochicus replied:
@TFK70 

> Задача не понятна или готовый код нужен?

Да, сформулируй как оно должно быть выполнено.  Что можно использовать, а что нельзя (ещё бы желательно почему). Подходит ли текущая верстка или какие то фрагменты требуют переписывания.  Ну, как минимум, обозначай проблему.  Нужно понимание в каком направлении двигаться. Например, сейчас, я абсолютно уверен, что не могу использовать `height` в белом блоке, или в `column` , потому что вылезут серьезные проблемы с контентом. Но может быть я ошибаюсь, и мы об этом ещё долго не узнаем, пока я просто закидываю варианты решений, игнорирующие задание высоты.

TFK70 replied:
@gochicus ну так ты меня сейчас просишь за тебя код написать, так не пойдет
> Что можно использовать, а что нельзя

Я вообще не уверен что есть вещи которые "нельзя" использовать, основная суть в том что у всего есть свое назначение, и если мы, например, даем бекраунду высоту 50vh - это уже никакой не бекграунд

> Подходит ли текущая верстка или какие то фрагменты требуют переписывания

Это ты сам должен выяснить. Проект и верстка твои, необходимые знания у тебя есть. Я просто слежу чтобы в твоем коде не было явных антипаттернов (как с тем же бекграундом)

> Нужно понимание в каком направлении двигаться

Тут советую еще раз разбить твою верстку на компоненты: что чем должно являться, так же как мы в example приложении делали в самом начале. Разметь верстку на высоком уровне: где должен быть Column, где Box и тд,  как каждый компонент должен изменяться, потом сравни с тем что у тебя есть сейчас

> Например, сейчас, я абсолютно уверен, что не могу использовать height в белом блоке, или в column , потому что вылезут серьезные проблемы с контентом.

Вполне вероятно пункт выше в этом поможет, это стандартная ситуация когда твоя верстка тебя же и блокирует. Тут только один вариант - переверстать (если возможно, конечно)

gochicus committed:
refactor(web-diary): markup refactored

affects: @fragments/diary, @fragments/post-list, @practice/web-diary, @ui/list

ISSUES CLOSED: #74

TFK70 commented on practice/web-diary/fragments/post-list/src/PostList.tsx:
```diff
@@ -12,12 +13,19 @@ export const PostList = () => {
+    <Column
+      width='100%'
+      justifyContent='center'
+      height={[postData.length <= 3 ? '20%' : '100%', '100%', '100%']}
```
Это конечно интересно, но есть путь попроще

TFK70 commented on practice/web-diary/ui/list/src/List.tsx:
```diff
@@ -1,12 +1,10 @@
 export const List = ({ items, deleteItem, importantItem, likeItem, setPostData }) => {
   return (
-    <Column width='100%' justifyContent='center'>
+    <>
```
Да, возможно List как компонент и стоит обернуть снаружи во что-то, что будет регулировать его размеры, но здесь Column должен остаться, ведь без него это уже не лист, как минимум потому, что итемы будут располагаться горизонтально

gochicus replied:
@TFK70 ты смотрел верстку? Они корректно, вертикально располагаются.

TFK70 replied:
@gochicus да, я смотрел и всегда смотрю
Я к тому что если List будет без колонки то элементы будут в ряду. Это неправильно, тк ты у себя в приложении позиционируешь List как самостоятельную сущность

gochicus replied:
@TFK70 я вот сейчас вообще не понял идею. List - ui компонент для использования. То есть, использовать его можно хоть в ряд, хоть в колонку.

TFK70 commented on practice/web-diary/fragments/diary/src/Diary.tsx:
```diff
@@ -9,7 +9,7 @@ import { Box, Column, Layout, Row } from '@ui/layout'
+    <Background gradient='blueAndGreenGradient' height='100vh' width='100%'>
       <Column alignItems='center'>
         <Layout flexBasis={['0px', '50px', '300px']} />
         <Box backgroundColor='white' width={['100%', '400px', '400px']}>
```
А в чем проблема тут height регулировать?

gochicus replied:
@TFK70 В мобильной версии едет именно колонка с айтемами, если задать высоту колонке здесь или же блоку с белым цветом, то при загрузке большого числа постов они будут наезжать друг на друга, пропадут отступы, а при оверфлоу контент будет резаться.

gochicus committed:
refactor(web-diary): markup refactor

affects: @fragments/diary, @fragments/post-list, @practice/web-diary, @ui/list

ISSUES CLOSED: #74

gochicus committed:
refactor(web-diary): markup refactor

affects: @fragments/diary, @practice/web-diary

ISSUES CLOSED: #74

TFK70 commented on practice/web-diary/fragments/post-list/src/PostList.tsx:
```diff
@@ -12,12 +13,14 @@ export const PostList = () => {
-      items={filterData(postData, status, searchResult)}
-      setPostData={setPostData}
-    />
+    <Box width='100%' justifyContent='center'>
```
Row

gochicus committed:
style(web-diary): lint changes

ISSUES CLOSED: #74

TFK70 commented on practice/web-diary/fragments/post-сontrols/src/PostControls.tsx:
```diff
@@ -17,6 +17,7 @@ export const PostControls = () => {
   const [, setSearchValue] = useSearchValue()
   const [postData, setPostData] = usePostData()
   return (
+    <Box width='100%' justifyContent='center'>
```
Row

gochicus committed:
refactor(web-diary): markup refactor

affects: @fragments/post-list, @practice/web-diary

ISSUES CLOSED: #74

TFK70 commented on practice/web-diary/fragments/post-list/src/PostList.tsx:
```diff
@@ -13,14 +13,14 @@ export const PostList = () => {
   const [searchResult] = useSearchValue()
   return (
-    <Box width='100%' justifyContent='center'>
+    <Row width='100%' justifyContent='center'>
```
Нуу иии? `<Row width='100%'`?

gochicus committed:
refactor(web-diary): markup refactor

affects: @fragments/post-list, @practice/web-diary

ISSUES CLOSED: #74

TFK70 commented on practice/web-diary/fragments/diary/src/Diary.tsx:
```diff
@@ -1,35 +1,33 @@
-            <Layout flexBasis='15px' />
-          </Row>
+        <Layout flexBasis={['0px', '50px', '300px']} />
+        <Box backgroundColor='white' width={['100%', '400px', '400px']}>
```
У тебя вид с таблета должен быть такой же как с мобилки:

![image](https://user-images.githubusercontent.com/62806448/122361870-17a6e600-cf71-11eb-944c-bfe4d3956f5c.png)

Чтобы контент не уменьшался как сейчас

gochicus committed:
refactor(web-diary): markup refactor

affects: @fragments/diary, @practice/web-diary

ISSUES CLOSED: #74

TFK70 commented on practice/web-diary/fragments/post-сontrols/src/PostControls.tsx:
```diff
@@ -19,8 +19,8 @@ export const PostControls = () => {
-        <Row justifyContent='center'>
-          <Box width='60%'>
+        <Row justifyContent='space-between'>
+          <Row>
```
Не уверен что понимаю зачем здесь инпут оборачивать в Row

gochicus committed:
refactor(web-diary): excessive Row deleted

ISSUES CLOSED: #74

TFK70 commented on practice/web-diary/fragments/post-сontrols/src/PostControls.tsx:
```diff
@@ -20,16 +20,14 @@ export const PostControls = () => {
+            }}
+          />
           <Layout flexBasis='8px' />
           <Row justifyContent='flex-end'>
```
А вот здесь Row зачем? У тебя же там в таблетке инпуту прям плохо по ширине

gochicus committed:
refactor(web-diary): markup refactor

ISSUES CLOSED: #74

gochicus committed:
refactor(web-diary): global styles changed

affects: @fragments/diary, @practice/web-diary, @ui/background, @ui/theme

ISSUES CLOSED: #74

TFK70 reviewed (approved):
У тебя вид с таблета должен быть такой же как с мобилки:

![image](https://user-images.githubusercontent.com/62806448/122361870-17a6e600-cf71-11eb-944c-bfe4d3956f5c.png)

Чтобы контент не уменьшался как сейчас
